cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java

```diff
@@ -288,6 +288,15 @@ private boolean mergeHdr(Camera.Parameters params, Hdr oldHdr) {
         return false;
     }
 
+
+    @Override
+    void setAudio(Audio audio) {
+        if (mIsCapturingVideo) {
+            throw new IllegalStateException("Can't change audio while recording a video.");
```

Comment:
Can you make this just a no-op instead of throwing?
You can use the camera logger and add a `LOG.w("Changing audio mode while recording. Changes will take place starting from next video)`

Also, before doing anything, can you also check that audio != mAudio ?